### PR TITLE
TST: test against numpy nightly

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
First part of https://github.com/shapely/shapely/issues/1972, to start with testing running numpy 2.0.dev

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -122,7 +122,8 @@ jobs:
           python -m pip install --disable-pip-version-check --upgrade pip
           pip install --upgrade wheel setuptools
           if [ -z "${{ matrix.numpy }}" ]; then
-            pip install --upgrade --pre Cython numpy pytest pytest-cov coveralls;
+            pip install --upgrade --pre Cython pytest pytest-cov coveralls;
+            pip install --upgrade --pre --only-binary :all: -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
```
Comment (martinfleis) [inline .github/workflows/tests.yml:126]:
```suggestion
            pip install --upgrade --pre --only-binary :all: -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple numpy
```

Scipy channel is no longer in use, the latest numpy there is from 2023-06-16.

Comment (jorisvandenbossche) [reply .github/workflows/tests.yml:126]: I copied it from their docs ... but the page I landed on from googling is the one from 1.24 ;)

**Comment by jorisvandenbossche** [comment]
Seems to be all good!

**Comment by caspervdw** [comment]
Big relief that numpy 2.* “just works” for shapely!

**Pull request merged**